### PR TITLE
Use _ARRAY_DIMENSIONS to create shared, named dimensions in ZarrHeader

### PR DESCRIPTION
**Pull request by rschmunk**
## Description of Changes

This PR adds code to the ZarrHeader class to look for the xarray/geozarr _ARRAY_DIMENSIONS attribute and, if present, use it to name a variable's dimensions and to create shared dimensions with the variable's group.

This is a first effort to see how much code/effort was necessary and may not consider cases where this info should be explicitly be ignored, or if there are additional Exceptions that one needs to watch for.

## PR Checklist

- [ ] Link to any issues that the PR addresses
- [ ] Add labels
- [x] Open as a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/)
       until ready for review
- [x] Make sure GitHub tests pass
- [x] Mark PR as "Ready for Review"

### CODE REVIEW

**Comment by rschmunk** [comment]
One test failed, but is that just a timeout?

File: gradle/root/fatJars.gradle [hunk 1]
```diff
@@ -52,6 +52,8 @@ dependencies {
   netcdfAll project(':cdm:cdm-image')
   netcdfAll project(':cdm:cdm-radial')
   netcdfAll project(':cdm:cdm-misc')
+  netcdfAll project(':cdm:cdm-s3')
+  netcdfAll project(':cdm:cdm-zarr')
   netcdfAll project(':bufr')
```
Comment (haileyajohnson) [inline gradle/root/fatJars.gradle]: we're not ready to include these in the fatJar by default :)

**Review by haileyajohnson — CHANGES_REQUESTED** [review]
Eventually needs more unit tests for Zarr/NcZarr but happy to merge this now since zarr is still "experimental" atm (without the zarr modules added to the fat jar)

File: gradle/root/fatJars.gradle (hunk 1 above)
Comment (rschmunk) [reply gradle/root/fatJars.gradle]: @haileyajohnson, I know. Half the time when I'm pushing my copy of the repo from Xcode to Github, I forget to uncheck the box for that file.

Comment (haileyajohnson) [reply gradle/root/fatJars.gradle]: I figured :)

**Review by haileyajohnson — APPROVED** [review]

**Pull request merged**